src/sql/src/parsers/copy_parser.rs

```diff
@@ -39,10 +41,74 @@ impl<'a> ParserContext<'a> {
                     actual: self.peek_token_as_string(),
                 })?;
 
-        self.parser
-            .expect_keyword(Keyword::TO)
+        let direction = self
+            .parser
+            .expect_one_of_keywords(&[Keyword::TO, Keyword::FROM])
+            .context(error::SyntaxSnafu { sql: self.sql })?;
+
+        match direction {
+            Keyword::TO => self.parse_copy_table_to(table_name),
+            Keyword::FROM => self.parse_copy_table_from(table_name),
+            _ => unreachable!(),
+        }
+    }
+
+    fn parse_copy_table_from(&mut self, table_name: ObjectName) -> Result<CopyTable> {
+        let uri = self
+            .parser
+            .parse_literal_string()
+            .with_context(|_| error::UnexpectedSnafu {
+                sql: self.sql,
+                expected: "a uri",
+                actual: self.peek_token_as_string(),
+            })?;
+
+        let options = self
+            .parser
+            .parse_options(Keyword::WITH)
             .context(error::SyntaxSnafu { sql: self.sql })?;
 
+        // default format is parquet
+        let mut format = Format::Parquet;
+        let mut pattern = None;
+        for option in options {
+            match option.name.value.to_uppercase().as_str() {
+                "FORMAT" => {
+                    if let Value::SingleQuotedString(fmt_str) = option.value {
+                        format = Format::try_from(fmt_str)?;
+                    }
+                }
+                "PATTERN" => {
+                    if let Value::SingleQuotedString(fmt_str) = option.value {
+                        pattern = Some(fmt_str);
+                    }
+                }
+                &_ => unreachable!(),
+            }
+        }
+
+        let connection_options = self
+            .parser
+            .parse_options(Keyword::CONNECTION)
+            .context(error::SyntaxSnafu { sql: self.sql })?;
+        let connection = if connection_options.is_empty() {
+            None
+        } else {
+            let mut connection = HashMap::with_capacity(connection_options.len());
+            for option in connection_options {
+                if let Value::SingleQuotedString(fmt_str) = option.value {
+                    connection.insert(option.name.value.to_uppercase(), fmt_str);
+                }
+            }
+            Some(connection)
+        };
```

Comment:
```suggestion
        let connection = connection_options
            .into_iter()
            .filter_map(|option| {
                let Value::SingleQuotedString(fmt_str) = option.value else {return None};
                Some((option.name.value.to_uppercase(), fmt_str))
            })
            .collect();
```

Also I think `HashMap<String, String>` is enough,there's no need to use `Option<HashMap<String, String>>` to represent connection parameters, since an empty map can expression `Option::None`.